src/daft-micropartition/src/micropartition.rs

```diff
@@ -615,6 +616,7 @@ pub(crate) fn read_csv_into_micropartition(
 pub(crate) fn read_parquet_into_micropartition(
     uris: &[&str],
     columns: Option<&[&str]>,
+    schema: Option<SchemaRef>,
```

Comment:
Parquet reads differ a little from CSV reads: for Parquet, the file format itself contains a schema and thus no external schema information is required when reading that file.

Therefore for `read_parquet_into_micropartition`, we will probably not want to pass in the schema (unlike the CSV reads!)

Instead, we can let `read_parquet_into_micropartition` perform its own schema inference/data parsing, and then later on we can coerce the resultant `MicroPartition` into the inferred schema. The overall flow would look something like:

```rust
// Naively read Parquet file(s) into a MicroPartition, no schema coercion applied
// Note that this all happens lazily because of the nature of MicroPartitions
// being a lazy-loading abstraction
let mp = read_parquet_into_micropartition(...);

let applied_schema = mp.schema().apply(schema_hints);
let mp = mp.cast_to_schema(&applied_schema);
```

Reply:
ah got it, made this change in latest commit